src/plots/cartesian/axes.js

```diff
@@ -4523,3 +4574,38 @@ function setShiftVal(ax, axShifts) {
         axShifts[ax.overlaying][ax.side] :
         (ax.shift || 0);
 }
+
+/**
+ * Checks if the given period is at least the period described by the tickformat or larger. If that
+ * is the case, they are compatible, because then the tickformat can be used to describe the period.
+ * E.g. it doesn't make sense to put a year label on a period spanning only a month.
+ * @param {number} period in ms
+ * @param {string} tickformat
+ * @returns {boolean}
+ */
```

Comment:
Please add additional mocks by copying `date_axes`, `date_axes_period` and `date_axes_period2` as well as `date_axes-ticklabelstep` to test all these cases.
Thank you!

Reply:
to clarify, the new property only has an effect if minor ticks are actually visible. So for all of these except `date_axes_period2` it would just test that the property has no effect.

Reply:
Yes please duplicate those mocks and add minor ticks with this new option.
Thank you!